### PR TITLE
Bookmark Games

### LINKED ISSUE #594: Bookmark game
Add the possibility to bookmark a game.

Should be available in context menu (long press), in the future game list screen, and in a game screen.

### PR DESCRIPTION
**Pull request by ZTL-UwU**
closes #594

- [ ] Show original state
- [x] Context menu
- [x] Game history
- Game screens
    - [x] Archieved game screen
    - [x] Game screen
    - [x] Analysis screen

### CODE REVIEW

**Comment by ZTL-UwU** [comment]
Blocked by https://github.com/lichess-org/mobile/issues/594#issuecomment-2241508078

File: lib/src/model/game/game_repository.dart [hunk 1]
```diff
@@ -104,4 +104,19 @@ class GameRepository {
       mapper: LightArchivedGame.fromServerJson,
     );
   }
+
+  Future<void> bookmark(GameId id, {int v = -1}) async {
+    // if v is -1, toggle the bookmark value on server
+    // otherwise explicitly set the new value
```
Comment (ornicar) [inline lib/src/model/game/game_repository.dart:110]: If think `-1` is a bad idea, better to always explicitly set the new value, than to assume the current server state.

Comment (ZTL-UwU) [reply lib/src/model/game/game_repository.dart:110]: Yes, this a temporary solution. When the `bookmarked` field is added I will remove the `-1`.

**Comment by veloce** [comment]
@ZTL-UwU what is missing here to complete this feature? I think we can move on now since https://github.com/lichess-org/lila/issues/15773 has been merged.

Also can you please fix the conflicts? Thanks.

**Comment by ZTL-UwU** [comment]
Yes, we can move on now. I'm just a bit busy irl. I'll pick it up once I have some free time.

**Comment by julien4215** [comment]
@ZTL-UwU I merged main into your branch here #1302 

Do you still plan to work on the bookmark feature ? Otherwise I'd like to try to complete this PR

**Comment by ZTL-UwU** [comment]
> Do you still plan to work on the bookmark feature ? Otherwise I'd like to try to complete this PR

Thanks, go ahead!

**Pull request closed without merging**